index.js

```diff
@@ -108,6 +112,12 @@ function fastifyStatic (fastify, opts, next) {
           if (opts.list && dirList.handle(pathname, opts.list)) {
             return dirList.send({ reply, dir: dirList.path(opts.root, pathname), options: opts.list, route: pathname })
           }
+
+          // root paths left to try?
+          if (Array.isArray(rootPath) && rootPathOffset <= rootPath.length) {
```

Comment:
are you sure <= is right? if `rootPathOffset === rootPath.length` and we're calling `pumpSendToReply` with `rootPathOffset + 1`, aren't we exceeding the bounds of the `rootPath` array?

Reply:
Yes, you are right. The offset is always going to be one smaller than the length of the array. 

I can change it to:

```
if (Array.isArray(rootPath) && rootPathOffset < (rootPath.length - 1))
```
The only unique case would be when there is only one element in the array and we get the offset = 0 and the length = 0 (length - 1). `0 < 0` evaluates to false but it seems a little messy.

Reply:
what about `offset < length`?

Reply:
The problem is the condition check is at the end so we essentially have a do-while loop. `offset < length` will run the pump function on more time past the bounds of the array. If I move the increment before the check then it should work.